### PR TITLE
New ADR settings in the Console

### LINKED ISSUE #5628: Use new ADR API for ADR settings in the Console
### Summary

The Console should use the new ADR settings API (`mac_settings.adr`) instead of the legacy `mac_settings.use_adr` and `mac_settings.adr_margin` fields.

### Current Situation

The Console provides the `mac_settings.use_adr` and `mac_settings.adr_margin` settings to users.

The problem with the `use_adr` field is that it is inherently flawed. https://github.com/TheThingsNetwork/lorawan-stack/pull/5326#issuecomment-1074989674 goes into the history of this field but the TL;DR is that setting `use_adr` to `false` does not actually disable ADR in any way, it just enables the manual mode, which is not what most users expect to happen.

### Why do we need this? Who uses it, and when?

Any user that would like to use static ADR settings or would like to really disable ADR completely via the Console MAC settings form.

### Proposed Implementation

The new ADR settings API has the concept of modes:
1. Dynamic Mode:
- Equivalent to the old `use_adr` = `true` mode.
- Example update request:
```json
{
	"end_device": {
		"mac_settings": {
			"use_adr": null,
			"adr_margin": null,
			"adr": {
				"dynamic": {
					"margin": 15.0,
					"min_data_rate_index": null,
					"max_data_rate_index": null,
					"min_tx_power_index": null,
					"max_tx_power_index": null,
					"min_nb_trans": null,
					"max_nb_trans": null
				}
			}
		}
	},
	"field_mask": {
		"paths": ["mac_settings.use_adr", "mac_settings.adr_margin", "mac_settings.adr.mode.dynamic"]
	}
}
```
- All of the fields are nullable, and you may use the `mac_settings.adr_margin` as a default for `mac_settings.adr.dynamic.margin`.
- While it would be nice to render all of the fields, just show the `margin` for now and leave the rest to be configured via CLI.
2. Static Mode
- It instructs the end device to always use the same settings. 
- Example update request:
```json
{
	"end_device": {
		"mac_settings": {
			"use_adr": null,
			"adr_margin": null,
			"adr": {
				"static": {
					"data_rate_index": 0,
					"tx_power_index": 1,
					"nb_trans": 2
				}
			}
		}
	},
	"field_mask": {
		"paths": ["mac_settings.use_adr", "mac_settings.adr_margin", "mac_settings.adr.mode.static"]
	}
}
```
- All of the fields are required, and no defaults are to be provided - the user should input everything themselves.
- Field descriptions:
   - `data_rate_index` - The data rate index to be used by the end device for uplink transmission.
   - `tx_power_index` - The transmission power index to be used by the end device for uplink transmission.
   - `nb_trans` - The number of uplink transmissions of the same physical frame.
4. Disabled Mode
- Disables ADR for good.
- Example update request:
```json
{
	"end_device": {
		"mac_settings": {
			"use_adr": null,
			"adr_margin": null,
			"adr": {
				"disabled": {}
			}
		}
	},
	"field_mask": {
		"paths": ["mac_settings.use_adr", "mac_settings.adr_margin", "mac_settings.adr.mode.disabled"]
	}
}
```

### Contributing

- [X] I can help by doing more research.
- [ ] I can help by implementing the feature after the proposal above is approved.
- [X] I can help by testing the feature before it's released.

### Code of Conduct

- [X] I agree to follow TTN's [Community Code of Conduct](https://github.com/TheThingsNetwork/Manifesto/blob/master/CodeOfConduct.md).

### PR DESCRIPTION
**Pull request by ryaplots**
#### Summary

Closes #5628 

#### Changes

- Add new adr fields
- Add adr to setValues
- Update validation schema

Screenshot 2022-09-01 at 16 45 41
Screenshot 2022-09-01 at 16 45 48
Screenshot 2022-09-01 at 16 45 53

#### Testing

cypress tests

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [ ] Documentation: Relevant documentation is added or updated.
- [x] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: pkg/webui/console/components/mac-settings-section/index.js [hunk 1]
```diff
@@ -68,8 +68,10 @@ const m = defineMessages({
   classCTimeout: 'Class C timeout',
   maxDutyCycle: 'Maximum duty cycle',
   desiredMaxDutyCycle: 'Desired maximum duty cycle',
-  useAdr: 'Use adaptive data rate (ADR)',
   adrMargin: 'ADR margin',
+  adrUplinks: 'ADR uplinks per frame',
```
Comment (adriansmares) [inline pkg/webui/console/components/mac-settings-section/index.js]: I think `ADR number of transmissions` is better.

File: pkg/webui/console/components/mac-settings-section/index.js [hunk 2]
```diff
@@ -469,43 +461,71 @@ const MacSettingsSection = props => {
         />
       </Form.FieldContainer>
       <Form.Field
-        name="mac_settings.use_adr"
-        component={Checkbox}
-        label={m.useAdr}
-        onChange={handleUseAdrChange}
+        name="mac_settings.adr"
+        component={Radio.Group}
+        title={'Adaptive data rate (ADR)'}
```
Comment (adriansmares) [inline pkg/webui/console/components/mac-settings-section/index.js]: Shouldn't this be internationalized as well ?

File: pkg/webui/console/components/mac-settings-section/index.js [hunk 3]
```diff
@@ -469,43 +461,71 @@ const MacSettingsSection = props => {
...
         tooltipId={tooltipIds.ADR_USE}
-      />
-      {useAdr && (
+        encode={encodeAdrMode}
+        decode={decodeAdrMode}
+      >
+        <Radio label={'Dynamic mode'} value="dynamic" />
```
Comment (adriansmares) [inline pkg/webui/console/components/mac-settings-section/index.js]: These too.

File: pkg/webui/console/components/mac-settings-section/index.js (hunk 2 above)
Comment (ryaplots) [reply pkg/webui/console/components/mac-settings-section/index.js]: True, I overlooked these. Will fix

**Review by adriansmares — COMMENTED** [review]
The i18n looks good now, but I don't see the option for the disabled ADR mode.

File: pkg/webui/console/components/mac-settings-section/index.js [hunk 4]
```diff
@@ -469,43 +465,71 @@ const MacSettingsSection = props => {
...
+        title={m.adrAdaptiveDataRate}
         tooltipId={tooltipIds.ADR_USE}
-      />
-      {useAdr && (
+        encode={encodeAdrMode}
+        decode={decodeAdrMode}
+      >
+        <Radio label={m.adrDynamic} value="dynamic" />
+        <Radio label={m.adrStatic} value="static" />
+        <Radio label={m.adrDisabled} value="disabled" />
```
Comment (ryaplots) [inline pkg/webui/console/components/mac-settings-section/index.js]: here

**Review by adriansmares — APPROVED** [review]

File: pkg/webui/console/components/mac-settings-section/index.js [hunk 5]
```diff
@@ -469,43 +465,71 @@ const MacSettingsSection = props => {
...
+      </Form.Field>
+      {isDynamicAdr && (
+        <Form.Field
+          title={m.adrMargin}
+          name="mac_settings.adr.dynamic.margin"
+          component={Input}
+          type="number"
+          tooltipId={tooltipIds.ADR_MARGIN}
+          inputWidth="xs"
+          append="dB"
```
Comment (kschiffer) [inline pkg/webui/console/components/mac-settings-section/index.js:488]: Any min/max we can use here @adriansmares ?

File: pkg/webui/console/views/device-general-settings/network-server-form/validation-schema.js [hunk 6]
```diff
@@ -343,23 +343,38 @@ const validationSchema = Yup.object()
...
+              return Yup.object().shape({
+                static: Yup.object().shape({
+                  data_rate_index: Yup.number().nullable(),
+                  tx_power_index: Yup.number().nullable(),
+                  nb_trans: Yup.number().nullable(),
+                }),
+              })
+            } else if (value && 'disabled' in value) {
+              return Yup.object().shape({
+                disabled: Yup.object().shape({}),
```
Comment (kschiffer) [inline pkg/webui/console/views/device-general-settings/network-server-form/validation-schema.js]:
```suggestion
                disabled: Yup.object().shape({}).noUnknown(),
```

**Review by kschiffer — COMMENTED** [review]
Please always add screenshots to PRs with visual changes.

File: pkg/webui/console/components/mac-settings-section/index.js (hunk 5 above)
Comment (adriansmares) [reply pkg/webui/console/components/mac-settings-section/index.js:488]: No, but having something like [-100, 100] as a Console sanity check is fine. The backend endures any value.

In general the indices (the settings in static and dynamic) can now be validated using https://github.com/TheThingsNetwork/lorawan-stack/issues/5403 (you can even render only valid options if you want).

File: pkg/webui/console/components/mac-settings-section/index.js [hunk 7]
```diff
@@ -68,8 +68,14 @@ const m = defineMessages({
...
   desiredMaxDutyCycle: 'Desired maximum duty cycle',
-  useAdr: 'Use adaptive data rate (ADR)',
   adrMargin: 'ADR margin',
+  adrUplinks: 'ADR number of transmissions',
+  adrAdaptiveDataRate: 'Adaptive data rate (ADR)',
+  adrDataRate: 'ADR data rate index',
+  adrTransPower: 'ADR transmission power index',
+  adrDynamic: 'Dynamic mode',
+  adrStatic: 'Static mode',
+  adrDisabled: 'Disabled',
```
Comment (kschiffer) [inline pkg/webui/console/components/mac-settings-section/index.js]:
```suggestion
```
Don't add duplicate messages for things that can already be found in `sharedMessages`.

**Review by kschiffer — APPROVED** [review]

**Pull request merged**